stdlib/public/core/FloatingPointTypes.swift.gyb

```diff
@@ -459,7 +459,13 @@ extension ${Self}: BinaryFloatingPoint {
     precondition(payload < ${Self}._quietNaNMask,
                  "NaN payload is not encodable.")
     var significand = payload
-    if !signaling { significand |= ${Self}._quietNaNMask }
+    if signaling {
+      precondition(payload != 0,
+                   "Signaling NaN must have non-zero payload")
+    }
+    else {
```

Comment:
The usual code style in the library is to put both braces on the same line, `} else {`.